### PR TITLE
[CIRCLE-20634] Handle Gitlab push hooks

### PR DESCRIPTION
**Pull request by conormcd**
Add some very simple handling and verification of Gitlab push hooks.

Also add some trivial tests for the rest of the codebase.

### CODE REVIEW

**Comment by polymeris** [comment]
Did we decide to follow the Google Java styleguide? If so, the package names with underscores do not https://google.github.io/styleguide/javaguide.html#s5.2.1-package-names

**Comment by conormcd** [comment]
> Did we decide to follow the Google Java styleguide? If so, the package names with underscores do not https://google.github.io/styleguide/javaguide.html#s5.2.1-package-names

Ah, good spot! The lint phase of the CI configuration should have caught this. I will change this in a subsequent PR, given the sweeping nature of such a change. Any suggestions?

File: src/test/java/com/circleci/gitlab_single_org_connector/ConnectorApplicationTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+class ConnectorApplicationTest {
+  private final Environment environment = mock(Environment.class);
+  private final JerseyEnvironment jersey = mock(JerseyEnvironment.class);
+  private final HealthCheckRegistry healthCheckRegistry = mock(HealthCheckRegistry.class);
+  private final MetricRegistry metricRegistry = mock(MetricRegistry.class);
+  private final ConnectorApplication application = new ConnectorApplication();
+  private final ConnectorConfiguration config = new ConnectorConfiguration();
+
+  @BeforeEach
+  void setup() throws Exception {
```
Comment (polymeris) [inline src/test/java/com/circleci/gitlab_single_org_connector/ConnectorApplicationTest.java]: I see the test classes in general are littered with `throws Exception` due to mappers :( ... but here it shouldn't be necessary.

File: src/main/java/com/circleci/gitlab_single_org_connector/ConnectorApplication.java [hunk 2]
```diff
@@ -40,7 +38,7 @@ private void maybeConfigureStatsdMetrics(ConnectorConfiguration config, MetricRe
    */
   public void run(ConnectorConfiguration config, Environment environment) throws Exception {
```
Comment (polymeris) [inline src/main/java/com/circleci/gitlab_single_org_connector/ConnectorApplication.java]: That `Exception` is inherited from the parent, but we could (should?) remove it.

File: src/test/java/com/circleci/gitlab_single_org_connector/ConnectorApplicationTest.java (hunk 1 above)
Comment (conormcd) [reply src/test/java/com/circleci/gitlab_single_org_connector/ConnectorApplicationTest.java]: Good point. Will fix.

File: src/main/java/com/circleci/gitlab_single_org_connector/ConnectorApplication.java (hunk 2 above)
Comment (conormcd) [reply src/main/java/com/circleci/gitlab_single_org_connector/ConnectorApplication.java]: Done!

**Comment by conormcd** [comment]
I'm going to merge this and then submit a different PR for the namespace change.

**Comment by conormcd** [comment]
https://github.com/circleci/gitlab-single-org-connector/pull/3 does the package rename.

**Pull request merged**